src/features/basic_spa/aggregate/index.js

```diff
@@ -0,0 +1,92 @@
+
+    this.harvestTimeSeconds = getConfigurationValue(agentIdentifier, 'spa.harvestTimeSeconds') || 10
+
+    this.initialPageLoadInteraction = new InitialPageLoadInteraction(this.agentIdentifier)
+    this.initialPageLoadInteraction.on('finished', ixn => {
+      if (!ixn) return
+      this.interactionsToHarvest.push(ixn)
+      this.scheduler.scheduleHarvest(0.1)
+    })
+
+    this.interactionInProgress = null
+    this.interactionsToHarvest = []
+    this.interactionsSent = []
+
+    this.blocked = false
+
+    // const tracerEE = this.ee.get('tracer') // used to get API-driven interactions
+
+    this.scheduler = new HarvestScheduler('events', {
+      onFinished: this.onHarvestFinished.bind(this),
+      retryDelay: this.harvestTimeSeconds
+    }, { agentIdentifier, ee: this.ee })
+    this.scheduler.harvest.on('events', this.onHarvestStarted.bind(this))
+
+    registerHandler('newInteraction', (timestamp, trigger, category) => this.startInteraction({ category, trigger, startedAt: timestamp }), this.featureName, this.ee)
+    registerHandler('newURL', (timestamp, url, type) => this.interactionInProgress?.updateHistory(timestamp, url), this.featureName, this.ee)
+    registerHandler('newDom', timestamp => this.interactionInProgress?.updateDom(timestamp), this.featureName, this.ee)
+
+    this.drain()
+  }
+
+  onHarvestStarted (options) {
+    if (this.interactionsToHarvest.length === 0 || this.blocked) return {}
+    const ixn = this.interactionsToHarvest.shift()
+    const payload = `bel.7;${ixn.serialize('bel')}`
+
+    this.interactionsSent.push(ixn)
+
+    if (this.interactionsToHarvest.length) this.scheduler.scheduleHarvest(0.1)
+
+    return { body: { e: payload } }
+  }
+
+  onHarvestFinished (result) {
+    if (result.sent && result.retry && this.interactionsSent.length > 0) {
+      this.interactionsSent.forEach((interaction) => {
+        this.interactionsToHarvest.unshift(interaction)
+      })
+      this.interactionsSent = []
+    }
+  }
+
+  startInteraction ({ trigger, category, startedAt }) {
+    console.log('START IXN', trigger, category, startedAt)
+    this.interactionInProgress?.cancel()
+    this.interactionInProgress = new Interaction(this.agentIdentifier)
+    this.interactionInProgress.on('finished', this.completeInteraction.bind(this))
+    this.interactionInProgress.on('cancelled', this.cancelInteraction.bind(this))
+    if (trigger) this.interactionInProgress.trigger = trigger
+    if (category) this.interactionInProgress.category = CATEGORY.ROUTE_CHANGE
+    if (startedAt) this.interactionInProgress.start = startedAt
+  }
+
+  cancelInteraction () {
+    this.interactionInProgress = null
+  }
+
+  completeInteraction (ixn) {
+    if (!ixn) return
+    this.interactionsToHarvest.push(ixn)
+    this.interactionInProgress = null
+    this.scheduler.scheduleHarvest(0.1)
+  }
+
+  hasInteraction ({ timestamp }) {
+    if (!timestamp) return { interaction: undefined }
```

Comment:
I think this is meant to return `interactions`? Can you align function name & property singular or plural?